ansible_runner/runner_config.py

```diff
@@ -181,9 +181,10 @@ def prepare(self):
 
         # write the SSH key data into a fifo read by ssh-agent
         if self.ssh_key_data:
-            self.ssh_key_path = os.path.join(self.artifact_dir, 'ssh_key_data')
-            open_fifo_write(self.ssh_key_path, self.ssh_key_data)
-            self.command = self.wrap_args_with_ssh_agent(self.command, self.ssh_key_path)
+            for key, value in enumerate(self.ssh_key_data):
```

Comment:
Something that we do with some of the other options is test to see if this is a list and if it is then we can enumerate over it, otherwise treat it as a single entity. That may be the best way to support backwards compatibility on the interface here.

Otherwise we'd end up breaking existing consumers.

Reply:
@matburt I also changed the single case (the current case) to be inside of a list (https://github.com/ansible/ansible-runner/pull/365/files#diff-c62b92cfe059167de0305220ea16db97R292). If we have a list in both cases, we can have the backwards compatibility without needing additional checks, which makes the code at large more obscure.